http/httpclient/src/main/java/com/okta/commons/http/httpclient/HttpClientRequestExecutor.java

```diff
@@ -263,17 +260,114 @@ private HttpHeaders getHeaders(HttpResponse response) {
         return headers;
     }
 
-    private static int parseConfigValue(String key, int defaultValue, String warning) {
+    private int getRequestExecutorParam(String key, String warning, int defaultValue) {
+        String configuredValueString = this.requestExecutorParams.get(key);
+        try {
+            if (configuredValueString != null) {
+                return Integer.parseInt(configuredValueString);
+            }
+        } catch (NumberFormatException nfe) {
+            log.warn("{}: {}. Using default: {}.",
+                warning,
+                configuredValueString,
+                defaultValue,
+                nfe);
+        }
+        return defaultValue;
+    }
 
-        int configuredValue = defaultValue;
-        String configuredValueString = System.getProperty(key);
-        if (configuredValueString != null) {
-            try {
-                configuredValue = Integer.parseInt(configuredValueString);
-            } catch (NumberFormatException nfe) {
-                log.warn("{}: {}. Using default: {}.", warning, configuredValueString, defaultValue, nfe);
+    public int getMaxConnectionPerRoute() {
+        return getRequestExecutorParam(
+            "maxPerRoute",
+            "Bad max connection per route value",
+            DEFAULT_MAX_CONNECTIONS_PER_ROUTE
+        );
+    }
+
+    public int getMaxConnectionTotal() {
+        return getRequestExecutorParam(
+            "maxTotal",
+            "Bad max connection total value",
+            DEFAULT_MAX_CONNECTIONS_TOTAL
+        );
+    }
+
+    public int getMaxConnectionInactivity() {
+        return getRequestExecutorParam(
+            "validateAfterInactivity",
+            "Invalid max connection inactivity validation value",
+            DEFAULT_CONNECTION_VALIDATION_INACTIVITY
+        );
+    }
+
+    public int getConnectionTimeToLive() {
+        return getRequestExecutorParam(
+            "timeToLive",
+            "Invalid connection time to live value",
+            DEFAULT_CONNECTION_TIME_TO_LIVE
+        );
+    }
```

Comment:
These likely should not be public, make them package-private if needed for tests, otherwise private.

You might be able to move all of this logic to the constructor to avoid need to have member methods only called from the constructor.

Reply:
fixed